html/cross-origin-opener-policy/resources/coop-coep.py

```diff
@@ -21,13 +21,41 @@ def main(request, response):
 <script src="/common/get-host-info.sub.js"></script>
 <iframe></iframe>
 <script>
-  const navigate = new URL(location).searchParams.get("navigate");
-  if (navigate !== null) {
-    self.location = navigate;
+  const params = new URL(location).searchParams;
+  const navHistory = params.get("navHistory");
+  const avoidBackAndForth = params.get("avoidBackAndForth");
+  const navigate = params.get("navigate");
+  const openerDOMAccess = params.get("openerDOMAccess");
+  // Need to wait until the page is fully loaded before navigating
+  // so that it creates a history entry properly.
+  async function fullyLoaded() {
+    return new Promise((resolve, reject) => {
+      addEventListener('load', () => {
+        requestAnimationFrame(() => {
+          requestAnimationFrame(() => {
+            resolve();
+          });
+        });
+      });
+    })
+  }
+  if (navHistory !== null) {
+    fullyLoaded().then(() => {
+      history.go(Number(navHistory));
+    });
+  } else if (navigate !== null && (history.length === 1 || !avoidBackAndForth)) {
```

Comment:
`avoidBackAndForth` doesn't seem necessary; I think the `history.length === 1` condition is sufficient.

Reply:
It is, but it prevents other (future) tests from doing multiple chained navigations.